src/clipboard_handlers/borders.ts

```diff
@@ -0,0 +1,105 @@
+import { pastedZones } from "../helpers/clipboard/clipboard_helpers";
+import {
+  Border,
+  CellPosition,
+  ClipboardData,
+  ClipboardOptions,
+  ClipboardPasteTarget,
+  HeaderIndex,
+  UID,
+  Zone,
+} from "../types";
+import { ClipboardHandler } from "./abstract_clipboard_handler";
+
+type ClipboardContent = {
+  borders: (Border | null)[][];
+  isCutOperation: boolean;
+};
+
+export class BorderClipboardHandler extends ClipboardHandler<ClipboardContent> {
+  copy(isCutOperation: boolean, data: ClipboardData): ClipboardContent | undefined {
+    const sheetId = this.getters.getActiveSheetId();
+    if (!("zones" in data) || data.zones.length === 0) {
+      return;
+    }
+    const { rowsIndex, columnsIndex } = data;
+    const borders: (Border | null)[][] = [];
+
+    for (const row of rowsIndex) {
+      const bordersInRow: (Border | null)[] = [];
+      for (const col of columnsIndex) {
+        const position = { col, row, sheetId };
+        bordersInRow.push(this.getters.getCellBorder(position));
+      }
+      borders.push(bordersInRow);
+    }
+    return { borders, isCutOperation };
+  }
+
+  paste(
+    target: ClipboardPasteTarget,
+    content: ClipboardContent,
+    options?: ClipboardOptions | undefined
+  ) {
+    if (!content) {
+      return;
+    }
+    const sheetId = this.getters.getActiveSheetId();
+    if (options?.pasteOption === "asValue") {
+      return;
+    }
+    if (!("borders" in content)) {
+      return;
+    }
+    if (!("zones" in target) || !target.zones.length) {
+      return;
+    }
+    const zones = target.zones;
+    if (!content.isCutOperation) {
+      this.pasteFromCopy(sheetId, zones, content);
+    } else {
+      const { left, top } = zones[0];
+      this.pasteZone(sheetId, left, top, content);
+    }
+  }
+
+  private pasteFromCopy(sheetId: UID, target: Zone[], content: any) {
+    if (target.length === 1) {
+      const height = content.borders.length;
+      const width = content.borders[0].length;
+      const pasteZones = pastedZones(target, width, height);
+      for (const zone of pasteZones) {
+        this.pasteZone(sheetId, zone.left, zone.top, content);
+      }
+    } else {
+      for (const zone of target) {
+        for (let col = zone.left; col <= zone.right; col++) {
+          for (let row = zone.top; row <= zone.bottom; row++) {
+            this.pasteZone(sheetId, col, row, content);
+          }
+        }
+      }
+    }
```

Comment:
Do you think this logic can be factorized and handled directly in the clipboard ? My idea would be to make the handlers.paste only paste once, but eh Clipboard would call multiple time handler.paste, once for every repeated zone.

Or if it's not possible/ugly to put that in the clipboard, at least have a helper `pasteFromCopy(pasteZone: Function, target, width, height)`, I feel like every handler have the exact same loop.

Actually I don't think the Filters have this repeating behaviour ATM, but I'm pretty sure it's more of an oversiht thatn anything else